androidTest/java/com/owncloud/android/authentication/SAMLAuthenticatorActivityTest.java

```diff
@@ -0,0 +1,290 @@
+    private static final String LOG_TAG = "LoginSuiteSAML";
+
+    private Context targetContext = null;
+    private String testServerURL = null;
+    private String testUser = null;
+    private String testPassword = null;
+    private String webViewUsernameId = null;
+    private String webViewPasswordId = null;
+    private String webViewSubmitXPath = null;
+
+    public enum ServerType {
+        /*
+         *  Server with trusted certificate
+         */
+        TRUSTED(1),
+
+        /*
+         * Sever with non-trusted certificate
+         */
+        NON_TRUSTED(2);
+
+        private final int status;
+
+        ServerType (int status) {
+            this.status = status;
+        }
+
+        public int getStatus() {
+            return status;
+        }
+
+        public static ServerType fromValue(int value) {
+            switch (value) {
+                case 1:
+                    return TRUSTED;
+                case 2:
+                    return NON_TRUSTED;
+            }
+            return null;
+        }
+
+    }
+
+    public ServerType servertype;
+
+    @Rule
+    public ActivityTestRule<AuthenticatorActivity> mActivityRule = new ActivityTestRule<AuthenticatorActivity>(
+            AuthenticatorActivity.class) {
+        @Override
+        protected Intent getActivityIntent() {
+
+            targetContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext();
+            Intent result = new Intent(targetContext, AuthenticatorActivity.class);
+            result.putExtra(EXTRA_ACTION, AuthenticatorActivity.ACTION_CREATE);
+            result.putExtra(EXTRA_ACCOUNT, "");
+            return result;
+        }
+    };
+
+    @Before
+    public void init() {
+        Bundle arguments = InstrumentationRegistry.getArguments();
+
+        testUser = arguments.getString("TEST_USER");
+        testPassword = arguments.getString("TEST_PASSWORD");
+        testServerURL = arguments.getString("TEST_SERVER_URL");
+        servertype = ServerType.fromValue(Integer.parseInt(arguments.getString("TRUSTED")));
+        webViewUsernameId = arguments.getString("TEST_USERNAME_ID");
+        webViewPasswordId = arguments.getString("TEST_PASSWORD_ID");
+        webViewSubmitXPath = arguments.getString("TEST_SUBMIT_XPATH");
+
+        // UiDevice available form API level 17
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+            UiDevice uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+            /*Point[] coordinates = new Point[4];
+            coordinates[0] = new Point(248, 1020);
+            coordinates[1] = new Point(248, 429);
+            coordinates[2] = new Point(796, 1020);
+            coordinates[3] = new Point(796, 429);*/
+            try {
+                if (!uiDevice.isScreenOn()) {
+                    uiDevice.wakeUp();
+                    //uiDevice.swipe(coordinates, 10);
```

Comment:
Code to simulate a swipe on the device. It can be useful... for the moment i will not remove it. Comment added